src/Features/CSharp/Portable/CodeRefactorings/NamedArguments/CSharpAddNamedArgumentsCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,188 @@
+                    var arrayCreation = WrapInArrayCreation(expressions);
+                    var namedArgument = Argument(arrayCreation).WithNameColon(NameColon(parameter.Name));
+                    yield return namedArgument;
+                    yield break;
+                }
+                else
+                {
+                    yield return argument.WithNameColon(NameColon(parameter.Name));
+                }
+            }
+        }
+
+        private static ImplicitArrayCreationExpressionSyntax WrapInArrayCreation(IEnumerable<ExpressionSyntax> expressions)
+        {
+            var initializer = InitializerExpression(SyntaxKind.ArrayInitializerExpression, SeparatedList(expressions));
+            return ImplicitArrayCreationExpression(initializer);
+        }
+
+        protected override bool IsCandidate(SyntaxNode node)
+        {
+            if (node == null)
+            {
+                throw new ArgumentNullException(nameof(node));
+            }
+
+            switch (node.Kind())
+            {
+                case SyntaxKind.InvocationExpression:
+                case SyntaxKind.ElementAccessExpression:
+                case SyntaxKind.ObjectCreationExpression:
+                case SyntaxKind.BaseConstructorInitializer:
+                case SyntaxKind.ThisConstructorInitializer:
+                    return true;
+                default:
+                    return false;
+            }
+        }
+
+        protected override bool TryGetArguments(SyntaxNode node, SemanticModel semanticModel, out IList<ArgumentSyntax> arguments)
+        {
+
+            if (node == null)
+            {
+                throw new ArgumentNullException(nameof(node));
+            }
+
+            if (semanticModel == null)
+            {
+                throw new ArgumentNullException(nameof(semanticModel));
+            }
+
+            switch (node.Kind())
+            {
+                case SyntaxKind.InvocationExpression:
+                    {
+                        var syntaxNode = (InvocationExpressionSyntax)node;
+
+                        // Skip nameof
+                        var symbol = semanticModel.GetSymbolInfo(syntaxNode).Symbol;
+                        if (symbol == null)
+                        {
+                            goto default;
+                        }
+
+                        arguments = syntaxNode.ArgumentList.Arguments.ToList();
+                        return true;
+                    }
+
+                case SyntaxKind.ElementAccessExpression:
+                    {
+                        var syntaxNode = (ElementAccessExpressionSyntax)node;
+
+                        // Array access is not allowed to contain named arguments
+                        var type = semanticModel.GetTypeInfo(syntaxNode.Expression).Type;
+                        if (type.TypeKind == TypeKind.Array)
+                        {
+                            goto default;
```

Comment:
I saw it in the codebase, thought it was ok to `goto` switch cases. Will change.